sosia/classes/original.py

```diff
@@ -1,5 +1,7 @@
 """Main module of `sosia` containing the `Original` class."""
 
+from __future__ import annotations
+from typing import Iterable, List, Optional, Union
```

Comment:
Have a look at my last comment regarding the `Iterable`.

Reply:
Done. Let's use `collections.abc` for `Iterable` and `Set`

Reply:
Actually let's reverse that. `typing` is the right choice for type hints; `collections.abc` is more for tests.